### PR TITLE
Fixed scrolling for chat

### PR DESCRIPTION
**Pull request by vishwjeet-pawar03**
## Description

[Provide a brief description of the changes in this PR]

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Is there an addition/change in API Request, Response? If yes, documentation is required)
- [ ] Performance improvement
- [ ] Code refactoring
- [ ] Security fix

## Related Issues

- Fixes #[issue number]
- Closes #[issue number]
- Related to #[issue number]

## How Has This Been Tested?

[Describe the tests you ran to verify your changes. Provide instructions so we can reproduce.]

### Test Configuration

- [ ] Unit tests pass
- [ ] Integration tests pass
- [ ] Manual testing completed
- [ ] Cross-browser testing (if applicable)
- [ ] Mobile responsiveness tested (if applicable)

## Core Functionality Testing

Please confirm that the following core functionalities are working as expected:

- [ ] **Search capabilities** are working correctly
- [ ] **Knowledge search** is functioning properly
- [ ] **Connector indexing** is working as expected
- [ ] **Citations** are displaying and linking correctly
- [ ] **Documentation** is updated at https://docs.pipeshub.com/introduction

## What You Have Tested

Please describe what you have specifically tested:

- [ ] Feature works in development environment
- [ ] Feature works in staging environment
- [ ] Error handling scenarios tested
- [ ] Edge cases considered and tested
- [ ] Performance impact assessed
- [ ] Security implications reviewed

### Test Results

[Provide details of your test results here]

## Screenshots/Videos

**Before:**
[Attach screenshots/videos showing the current behavior]

**After:**
[Attach screenshots/videos showing the new behavior]

**Note:** Screenshots are required before merge for UI changes.

## Code Quality Checklist

- [ ] My code follows the project's coding standards
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Documentation

- [ ] Documentation has been updated (if applicable)
- [ ] API documentation updated (if applicable)
- [ ] README updated (if applicable)
- [ ] Changelog updated (if applicable)

## Security Considerations

- [ ] No sensitive data is exposed
- [ ] Input validation is implemented where necessary
- [ ] Authentication/authorization is properly handled
- [ ] No security vulnerabilities introduced

## Breaking Changes

- [ ] This PR introduces breaking changes
- [ ] Migration guide provided (if applicable)
- [ ] Version bump required

If breaking changes are introduced, please describe them here:

[Describe any breaking changes and how users should adapt]

## Performance Impact

- [ ] No performance impact
- [ ] Performance improved
- [ ] Performance impact assessed and acceptable

[Describe any performance implications]

## Dependencies

- [ ] No new dependencies added
- [ ] New dependencies are necessary and approved
- [ ] Dependencies updated and tested

List any new dependencies:
- [Dependency name and version]

## Deployment Notes

[Any special deployment considerations or steps]

## Checklist Before Merge

- [ ] All tests are passing
- [ ] Code review completed and approved
- [ ] Documentation updated and reviewed
- [ ] Screenshots/videos attached for UI changes
- [ ] Core functionality verified
- [ ] Security review completed (if applicable)
- [ ] Performance impact assessed
- [ ] Ready for production deployment

## Additional Notes

[Any additional information that would be helpful for reviewers]

---

**For Reviewers:**

Please ensure all checklist items are completed before approving this PR. Pay special attention to:
- Core functionality testing
- Security implications
- Performance impact
- Documentation completeness

### CODE REVIEW

**Comment by vishwjeet-pawar03** [comment]
@claude review this PR

File: frontend/app/(main)/chat/components/message-area/message-list.tsx [hunk 1]
```diff
@@ -544,6 +594,52 @@ export function MessageList() {
...
+        const dist = scrollHeight - scrollTop - clientHeight;
+        const maxST = Math.max(0, scrollHeight - clientHeight);
+        const flush = dist <= 4 || scrollTop >= maxST - 3;
+        if (flush) {
+          isScrolledUpRef.current = false;
+          lastTailSyncScrollTopRef.current = scrollTop;
+          requestAnimationFrame(() => {
+            if (!isStreamingRef.current || isScrolledUpRef.current) return;
+            syncStreamingMessageTailRef.current();
+          });
```
[the remark this answers is not in the document]
Comment (vishwjeet-pawar03) [reply frontend/app/(main)/chat/components/message-area/message-list.tsx]: Fixed

File: frontend/app/(main)/chat/components/message-area/message-list.tsx [hunk 2]
```diff
@@ -488,17 +470,85 @@ export function MessageList() {
...
+  // tail-sync position; resume only when flush with the doc bottom (tight
+  // `dist` or at max scroll). Resume MUST run before the opt-out check so a
+  // stale anchor cannot block “scroll back down to follow.” On resume, refresh
+  // the anchor and run tail-sync on rAF so the next chunk is not the first time
+  // we snap.
+  const STREAMING_RESUME_DIST_FLUSH_PX = 4;
+  const STREAMING_RESUME_AT_MAX_EPS_PX = 3;
+  const STREAMING_OPT_OUT_ANCHOR_PX = 4;
+  const IDLE_TAIL_BOTTOM_ZONE_PX = 80;
+  const IDLE_SCROLLED_UP_ZONE_PX = 96;
```
[the remark this answers is not in the document]
Comment (vishwjeet-pawar03) [reply frontend/app/(main)/chat/components/message-area/message-list.tsx]: Fixed

File: frontend/app/(main)/chat/components/message-area/message-list.tsx [hunk 3]
```diff
@@ -544,6 +594,52 @@ export function MessageList() {
...
+      if (!isStreamingRef.current) return;
+      if (ev.deltaY < 0) {
+        isScrolledUpRef.current = true;
+        return;
+      }
+      if (ev.deltaY > 0) {
+        const { scrollTop, scrollHeight, clientHeight } = el;
+        const dist = scrollHeight - scrollTop - clientHeight;
+        const maxST = Math.max(0, scrollHeight - clientHeight);
+        const flush = dist <= 4 || scrollTop >= maxST - 3;
```
[the remark this answers is not in the document]
Comment (vishwjeet-pawar03) [reply frontend/app/(main)/chat/components/message-area/message-list.tsx]: Fixed

File: frontend/app/(main)/chat/components/message-area/message-list.tsx [hunk 4]
```diff
@@ -544,6 +594,52 @@ export function MessageList() {
...
+        }
+      }
+    };
+    let touchScrollTop0 = 0;
+    const onTouchStart = () => {
+      touchScrollTop0 = el.scrollTop;
+    };
+    const onTouchEnd = () => {
+      if (!isStreamingRef.current) return;
+      if (el.scrollTop < touchScrollTop0 - 8) {
```
[the remark this answers is not in the document]
Comment (vishwjeet-pawar03) [reply frontend/app/(main)/chat/components/message-area/message-list.tsx]: Fixed

File: frontend/app/(main)/chat/components/message-area/message-list.tsx [hunk 5]
```diff
@@ -139,12 +139,19 @@ export function MessageList() {
...
   isStreamingRef.current = isStreaming;
-  // Timestamp (ms) of the last time we cleared the scroll lock by reaching
-  // the bottom. Used to implement a grace period during which the lock cannot
-  // be immediately re-engaged — prevents the race where streaming content
-  // grows scrollHeight an instant after the user reaches the bottom, then
-  // inertia scroll events see distFromBottom > 150 and wrongly re-lock.
+  // Timestamp (ms) of the last time we treated the viewport as pinned to the
+  // live tail (`distFromBottom` in the bottom zone). Crossing into the "reading
+  // up" zone within a short window after that is ignored so bounce / layout
+  // jitter does not flip `isScrolledUp` while we still intend to follow SSE.
```
[the remark this answers is not in the document]
Comment (vishwjeet-pawar03) [reply frontend/app/(main)/chat/components/message-area/message-list.tsx]: Fixed

**Review by abhishek-pipeshub — APPROVED** [review]
